src/Tizen.Diagnostics/Tizen.Diagnostics/Diagnostics.cs

```diff
@@ -0,0 +1,294 @@
+        internal DiagnosticsData(IntPtr handle)
+        {
+            _handle = handle;
+        }
+
+        /// <summary>
+        /// Finalizes an instance of the DiagnosticsData class.
+        /// </summary>
+        ~DiagnosticsData()
+        {
+            Dispose(false);
+        }
+
+        /// <summary>
+        /// Reads given number of bytes from diagnostics data.
+        /// </summary>
+        /// <since_tizen> 8 </since_tizen>
+        /// <param name="buffer">An array of bytes. When this method returns, the buffer contains the specified byte array with the values between offset and (offset + count - 1) replaced by the bytes read from the current source.</param>
+        /// <param name="offset">The zero-based byte offset in buffer at which to begin storing the data read from the current stream.</param>
+        /// <param name="count">The maximum number of bytes to be read.</param>
+        /// <returns>
+        /// The total number of bytes read into the buffer. This can be less than the number of bytes requested if that many bytes are not currently available, or zero (0) if the end of the stream has been reached.
+        /// </returns>
+        public int Read(byte[] buffer, int offset, int count)
+        {
+            return ReadUnsafe(buffer, offset, count, 0);
+        }
+
+        /// <summary>
+        /// Reads given number of bytes from diagnostics data.
+        /// </summary>
+        /// <since_tizen> 8 </since_tizen>
+        /// <param name="buffer">An array of bytes. When this method returns, the buffer contains the specified byte array with the values between offset and (offset + count - 1) replaced by the bytes read from the current source.</param>
+        /// <param name="offset">The zero-based byte offset in buffer at which to begin storing the data read from the current stream.</param>
+        /// <param name="count">The maximum number of bytes to be read.</param>
+        /// <param name="timeout">Timeout [ms] for reading requested number of bytes.</param>
+        /// <returns>
+        /// The total number of bytes read into the buffer. This can be less than the number of bytes requested if that many bytes are not currently available, or zero (0) if the end of the stream has been reached.
+        /// </returns>
+        public int Read(byte[] buffer, int offset, int count, int timeout)
+        {
+            return ReadUnsafe(buffer, offset, count, timeout);
+        }
+
+        // int Read(Byte[] buffer, int count)
+        private unsafe int ReadUnsafe(byte[] buffer, int offset, int count, int timeout)
+        {
+            var length = Convert.ToUInt32(count);
+            uint bytes_read = 0;
+
+            if (offset + count > buffer.Length)
+                throw new ArgumentException("The sum of offset and count is larger than the buffer length");
+
+            if (buffer == null)
+                throw new ArgumentNullException(nameof(buffer));
```

Comment:
buffer should be checked before use buffer.Length
```suggestion
            if (buffer == null)
                throw new ArgumentNullException(nameof(buffer));

            if (offset + count > buffer.Length)
                throw new ArgumentException("The sum of offset and count is larger than the buffer length");
```